kolibri/plugins/learn/assets/src/views/LibraryPage/PinnedNetworkResources.vue

```diff
@@ -1,26 +1,34 @@
 <template>
 
-  <KGrid>
+  <!-- eslint-disable vue/html-indent -->
+
+  <KGrid v-if="pinnedDevices !== null">
     <KGridItem
-      v-for="group in groups"
-      :key="group.id"
+      v-for="device in pinnedDevices"
+      :key="device.id"
     >
       <KGridItem>
         <h2>
-          <KIcon icon="device" />
-          <span class="device-name">{{ group.deviceName }}</span>
+          <KIcon :icon="getDeviceIcon(device)" />
+          <span class="device-name">{{ device.device_name }}</span>
         </h2>
       </KGridItem>
-      <KGridItem
-        v-for="item in group.content"
-        :key="item.id"
-        :layout="{ span: cardColumnSpan, alignment: 'auto' }"
-      >
-        <CardContent
-          :content="item.title"
-          :body="item.body"
-        />
-      </KGridItem>
+      <div class="card-layout">
+        <div class="cards">
+          <ChannelCardGroupGrid
+            data-test="channel-cards"
+            class="grid"
+            :contents="device.channels"
+            :isPinnedDevice="true"
+          >
+            <ExploreCard
+              :style="cardStyle"
+              class="card-main-wrapper"
+            />
```

Comment:
Yes, I think that generally speaking, using grid items within a grid ensures consistency, that's how grid components are intended to be used.

Reply:
It may be simpler to wrap the newly added slot in the `ChannelCardGroupGrid` in `KGridItem`:

```
<KGridItem>
  <slot></slot>
</KGridItem>
```

(however it depends on how that new slot was intended, I'm not familiar with the whole pull request)